### PR TITLE
Swaggerify /ban

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: api/client-server/v1/banning.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            type: object
+            example: |-
+              {
+                "reason": "Telling unfunny jokes",
+                "user_id": "@cheeky_monkey:matrix.org"
+              }
+            properties:
+              user_id:
+                type: string
+                description: The fully qualified user ID of the invitee.
```
Comment (Kegsay) [inline api/client-server/v1/banning.yaml]: "invitee"? Really? :stuck_out_tongue:

File: api/client-server/v1/banning.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+              reason:
+                type: string
+                description: The reason the user has been banned.
+            required: ["reason", "user_id"]
```
Comment (Kegsay) [inline api/client-server/v1/banning.yaml]: Is `reason` actually *required* for banning?

File: api/client-server/v1/banning.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      responses:
+        200:
+          description: The user has been kiced from and banned from re-joining the room.
```
Comment (Kegsay) [inline api/client-server/v1/banning.yaml]: s/kicked from and banned from/kicked and banned from/

File: api/client-server/v1/banning.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            application/json: |-
+              {}
+          schema:
+            type: object
+        403:
+          description: |-
+            You do not have permission to invite the user to the room. A meaningful ``errcode`` and description error text will be returned. Example reasons for rejections are:
+
+            - The inviter is not currently in the room.
+            - The inviter's power level is insufficient to invite users to the room.
```
Comment (Kegsay) [inline api/client-server/v1/banning.yaml]: This isn't an invite.

File: api/client-server/v1/banning.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    type: apiKey
+    description: The user_id or application service access_token
+    name: access_token
+    in: query
+paths:
+  "/rooms/{roomId}/ban":
+    post:
+      summary: Ban a user in the room.
+      description: |-
+        Ban a user in the room (and kick them, as a side effect).
```
Comment (Kegsay) [inline api/client-server/v1/banning.yaml]: And what if the user is not in the room? We should explicitly state what happens here.

File: api/client-server/v1/banning.yaml (hunk 2 above)
Comment (illicitonion) [inline api/client-server/v1/banning.yaml]: No. In fact, it's currently completely ignored by synapse. Yay. Removed.

File: api/client-server/v1/banning.yaml (hunk 1 above)
Comment (illicitonion) [inline api/client-server/v1/banning.yaml]: Done

File: api/client-server/v1/banning.yaml (hunk 3 above)
Comment (illicitonion) [inline api/client-server/v1/banning.yaml]: Done

File: api/client-server/v1/banning.yaml (hunk 4 above)
Comment (illicitonion) [inline api/client-server/v1/banning.yaml]: Done

File: api/client-server/v1/banning.yaml (hunk 5 above)
Comment (illicitonion) [inline api/client-server/v1/banning.yaml]: Done

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**